src/GraphQL/API/Enum.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DefaultSignatures #-}
+
+module GraphQL.API.Enum
+  ( GraphQLEnum(..)
+  ) where
+
+import Protolude hiding (Enum, U1, TypeError)
+import GraphQL.Internal.AST (Name, nameFromSymbol, NameError)
+import qualified GraphQL.Value as GValue
+import GHC.Generics (M1(..), Rep, Meta(..), C1, U1(..), D, (:+:)(..))
```

Comment:
We get many of these from Protolude: https://github.com/sdiehl/protolude/blob/39d56792ced9105e3019841f090bd55f3c12b19f/src/Protolude.hs#L362